src/lib/js/number.rs

```diff
@@ -0,0 +1,382 @@
+    Ok(to_value(this_fixed_num))
+}
+
+/// Number().toLocaleString()
+///
+/// https://tc39.es/ecma262/#sec-number.prototype.tolocalestring
+///
+/// Note that while this technically conforms to the Ecma standard, it does no actual
+/// internationalization logic.
+pub fn to_locale_string(this: &Value, _args: &[Value], _ctx: &mut Interpreter) -> ResultValue {
+    let this_num = to_number(this).to_num();
+    let this_str_num = format!("{}", this_num);
+    Ok(to_value(this_str_num))
+}
+
+/// Number().toPrecision(p)
+///
+/// https://tc39.es/ecma262/#sec-number.prototype.toprecision
+pub fn to_precision(this: &Value, args: &[Value], _ctx: &mut Interpreter) -> ResultValue {
+    println!("Number::to_precision()");
+    let this_num = to_number(this);
+    let _num_str_len = format!("{}", this_num.to_num()).len();
+    let _precision = match args.get(0) {
+        Some(n) => match n.to_int() {
+            x if x > 0 => n.to_int() as usize,
+            _ => 0,
+        },
+        None => 0,
+    };
+    // TODO: Implement toPrecision
+    unimplemented!();
+}
+
+/// Number().toString()
+///
+/// https://tc39.es/ecma262/#sec-number.prototype.tostring
+pub fn to_string(this: &Value, _args: &[Value], _ctx: &mut Interpreter) -> ResultValue {
+    Ok(to_value(format!("{}", to_number(this).to_num())))
+}
+
+/// Number().valueOf()
+///
+/// https://tc39.es/ecma262/#sec-number.prototype.valueof
+pub fn value_of(this: &Value, _args: &[Value], _ctx: &mut Interpreter) -> ResultValue {
+    Ok(to_number(this))
+}
+
+/// Create a new `Number` object
+pub fn create_constructor(global: &Value) -> Value {
+    let mut number_constructor = Object::default();
+    number_constructor.kind = ObjectKind::Function;
+
+    number_constructor.set_internal_method("construct", make_number);
+    number_constructor.set_internal_method("call", call_number);
+
+    let number_prototype = ValueData::new_obj(Some(global));
+
+    number_prototype.set_internal_slot("NumberData", to_value(0));
+
+    number_prototype.set_field_slice(
+        "toExponential",
+        to_value(to_exponential as NativeFunctionData),
+    );
+    number_prototype.set_field_slice("toFixed", to_value(to_fixed as NativeFunctionData));
+    number_prototype.set_field_slice(
+        "toLocaleString",
+        to_value(to_locale_string as NativeFunctionData),
+    );
+    number_prototype.set_field_slice("toPrecision", to_value(to_precision as NativeFunctionData));
+    number_prototype.set_field_slice("toString", to_value(to_string as NativeFunctionData));
+    number_prototype.set_field_slice("valueOf", to_value(value_of as NativeFunctionData));
+
+    let number = to_value(number_constructor);
+    number_prototype.set_field_slice("constructor", number.clone());
+    number.set_field_slice(PROTOTYPE, number_prototype);
+    number
+}
+
+/// Initalize the `Number` object on the global object
+pub fn init(global: &Value) {
+    global.set_field_slice("Number", create_constructor(global));
+}
+
```

Comment:
This logic should move to here:
https://github.com/jasonwilliams/boa/blob/master/src/lib/realm.rs#L57-L64 to match with the other objects or be gotten rid off